Improve search functionality by adding Search All & Ignore diacritics

MajdaSaid opened the pull request:
** Ignore diacritics in search dialog in text frame & story editor:
- now you can search for text with ignore diacritics & kashida option,
implemented by using ICU nonspacing-mark category.
- replace text also will consider  diacritics & kashida.
- fix search in linked frames.

** adding Search All functionality
Now you can search & replace in all text frames in scribus document. The
search algorithm is to start from selected text frame, if there is no
text frame selected select the first text frame in the current page, if
not, find the first text frame from next pages, if reaching the end of
the document start searching from the first page.

We add option to search only in current selected text frame.

Fahad-Alsaidi commented:
@MajdaSaid There is a bug when search text and only replace style such as font size. There is many spaces inserted and the style isn't applied.

Fahad-Alsaidi commented on scribus/scribus.cpp:
```diff
@@ -8456,12 +8456,77 @@ void ScribusMainWindow::EditTabs()
+	SearchReplace* dia = new SearchReplace(this, doc, currItem, true , SearchCur);
 	connect(dia, SIGNAL(NewFont(const QString&)), this, SLOT(SetNewFont(const QString&)));
 	connect(dia, SIGNAL(NewAbs(int)), this, SLOT(setAlignmentValue(int)));
+	connect(dia, SIGNAL(selectElementByItem(PageItem*,bool)), this, SLOT(selectItemsFromOutlines(PageItem*, bool)));
```
This is not needed anymore.

Fahad-Alsaidi commented on scribus/ui/search.h:
```diff
@@ -123,6 +125,7 @@ public slots:
 signals:
 	void NewFont(const QString&);
 	void NewAbs(int);
+	void selectElementByItem(PageItem *ite, bool single = true);
```
not needed anymore! please delete it.

khaledhosny commented:
The mark skipping seems to be based on an older version of the LibreOffice feature, I have recently updated this code to be simpler, future-proof and support more languages. See https://cgit.freedesktop.org/libreoffice/core/commit/?id=278eabab2b5bdc95a51d501fcdb46c216ded3baa

MajdaSaid committed:
Improve search functionality by adding Search All & Ignore diacritics

** Ignore diacritics in search dialog in text frame & story editor:
- now you can search for text with ignore diacritics & kashida option,
implemented by using ICU nonspacing-mark category.
- replace text also will consider  diacritics & kashida.
- fix search in linked frames.

** adding Search All functionality
Now you can search & replace in all text frames in scribus document. The
search algorithm is to start from selected text frame, if there is no
text frame selected select the first text frame in the current page, if
not, find the first text frame from next pages, if reaching the end of
the document start searching from the first page.

We add option to search only in current selected text frame.

MajdaSaid committed:
Style Replacment

MajdaSaid committed:
clean up

MajdaSaid committed:
facing link problem
